Fixes WQA Recipe - Nov 2020

Mranfevb92 opened the pull request:
Changes to deal with the issues pointed out in this Issue #17

Commit:
Revenue + Conversion Rate

* Fix to the the domains_proc model to be able to track more than one domain per site (to keep in consideration submodules with purchases)

* Removal of the blended_conversions fields on ga_stats and on following models.

Commit:
Update agg_traffic.sql

codingisforlosers reviewed (changes requested):
Thanks Andres, couple notes on these:

1) Left a comment on the issue, but for the GA domain / hostname issue, we should instead remove the `WHERE domain = url_domain` filter on `ga_proc`, and calculate the URL as `domain` + `landing_page_path` instead of `hostname` + `landing_page_path`.

2) Also left a comment on the issue, but for calculating blended conversion rates, we can take both routes (calculate in models but also Data Studio).  That way we'll have line-level details to calculate the actions, but when showing aggregates in Data Studio we'll re-calculate in aggregate vs taking an average.

Commit:
Revenue values, failing content and new page classification

Mranfevb92 commented:
Pushed a new commit. In this one I've: 

1. 
> Only thing we can really do there, is migrate to just rolling up stats for the landing_page_path (rather than concatenating the URL as hostname + landing_page_path). So URL could just be the site's root domain + landing_page_path from GA (of just landing_page_path in the event that the path contains the root domain, as is true for 28bysamwood).

Made this change to the `ga_proc`. I left in the previous changes I've made to the `domains_proc` model though. Let me know if you think we should remove those. They are working fine for now. 

2. 

> We can add a WHEN http_status_code = 200 condition here to solve the issue. We may want a more nuanced way of picking up if it's a new page, but I think sessions_30d equaling sessions from the last year is a decent proxy.

Made this change exactly on the a`ctions_herarchy`

3. For the failing_content category, aimed to use the logic we use on the OP models, so: 

* Calculated lag version of sessions_30d, transaction_30d, transaction_revenue_30d and goal_completions_all_goals_30d. All lag is to last 30 days. 

I've made the comparison: 

Rising Content: If the diff between monthly sessions is > 10%
Failing Content: If the diff between monthly sessions is < -10%

Let me know if you think we should shift that threshold: we could technically calculate a mid_diff if you think that is the best way

Also, I left the other categories in `content_action` as they were. But thinking that we may want to split `content_action` into two different metrics: 

1. The measure of rising content vs failing content
2. Another one for the proper content_actions? 

Also, maybe split rising content vs failing content by traffic but also by conversions/e-commerce conversions? Let me know if you think is a good possibility

codingisforlosers commented on models/metrics/actions/actions_proc.sql:
```diff
@@ -65,9 +255,8 @@ CASE WHEN http_status_code != 200 OR flag_paginated = 1 OR canonical_status = 'c
-	WHEN http_status_code = 200 AND transactions_30d > 0 and sessions_30d > med_sessions_30d and ecommerce_conversion_rate_mom_pct < total_organic_ecommerce_conversion_rate_mom_pct THEN 'Below-trend conversion rate'
-	WHEN http_status_code = 200 AND sessions_30d > med_sessions_30d and goal_conversion_rate_all_goals_mom_pct < total_organic_goal_conversion_rate_mom_pct THEN 'Below-trend conversion rate'
+	WHEN http_status_code = 200 AND sessions_30d != 0 AND sessions_30d_mom != 0 AND (sessions_30d - sessions_30d_mom)/sessions_30d > 0.1 THEN 'Rising content'
+	WHEN http_status_code = 200 AND sessions_30d != 0 AND sessions_30d_mom != 0 AND (sessions_30d - sessions_30d_mom)/sessions_30d < -0.1 THEN 'failing_content'
```
Let's call this status 'Falling content' rather than 'failing_content' (ie rising as opposite of falling).

codingisforlosers commented:
Thanks!  A couple notes:

> 
> > Only thing we can really do there, is migrate to just rolling up stats for the landing_page_path (rather than concatenating the URL as hostname + landing_page_path). So URL could just be the site's root domain + landing_page_path from GA (of just landing_page_path in the event that the path contains the root domain, as is true for 28bysamwood).
> 
> Made this change to the `ga_proc`. I left in the previous changes I've made to the `domains_proc` model though. Let me know if you think we should remove those. They are working fine for now.

In `ga_proc`, can't we just join `ga.account` to `domains_proc.google_analytics_account` (rather than joining on hostname = domain)?  That'd allow us to remove the added join in `domains_proc`, which could cause unforeseen issues down the line.


> 1. For the failing_content category, aimed to use the logic we use on the OP models, so:
> 
> * Calculated lag version of sessions_30d, transaction_30d, transaction_revenue_30d and goal_completions_all_goals_30d. All lag is to last 30 days.
> 
> I've made the comparison:
> 
> Rising Content: If the diff between monthly sessions is > 10%
> Failing Content: If the diff between monthly sessions is < -10%
> 
> Let me know if you think we should shift that threshold: we could technically calculate a mid_diff if you think that is the best way

I like that approach, left a comment above about the action name label being 'Falling content'.  

We may want to key it off of the site's overall growth rate though (ie 25% above / below total_organic_sessions_mom_pct), to account for seasonality.   There may be months where every page on the site loses traffic MoM (in January for a retail business, for example), so we still want to be able to differentiate which pages gained / lost traffic *relative to the the site as a whole*.

> 
> Also, I left the other categories in `content_action` as they were. But thinking that we may want to split `content_action` into two different metrics:
> 
> 1. The measure of rising content vs failing content
> 2. Another one for the proper content_actions?
> 
> Also, maybe split rising content vs failing content by traffic but also by conversions/e-commerce conversions? Let me know if you think is a good possibility

I agree on splitting into two cols - maybe one is `content_trajectory` (ie rising / falling), and the other is `content_action`?

Rising or falling isn't really an action recommendation, so I like the idea of splitting it out.

Commit:
domain_proc, search console proc, content_trajectory + content_action

* Adjusted the ga_proc so the join is through the ga account/ gsc account instead of by hostname.

* returned the domain_proc to its original version + an adjustment to calculate the lv field using the analytics and search console account on the partition.

*Adjustments to the search_console_url_proc and search_console_keyword_proc so they match the same as the ga_proc with the domain_proc

*created action_trajectory and modified content_action accordingly.

* added action_trajectory to all subsequent models after actions_proc

Commit:
Update search_console_keyword_proc.sql

Mranfevb92 committed:
Merge branch 'master' into Changes_Nov2020

codingisforlosers commented on models/visualization/actions_count.sql:
```diff
@@ -53,7 +53,7 @@ FROM (
 	FROM {{ ref('actions_hierarchy') }}
 	WHERE content_action != '' AND content_action is not null	
-	AND content_action not like 'rising%'
+	AND content_trajectory not like 'rising%'
```
@Mranfevb92 shouldn't this filter be `content_action != ''`?  Goal is to filter for non-blank content actions (the previous filter was due to 'Rising content' not being a true action.

Mranfevb92 replied:
Oh ok, though that the purpose was to keep only the pages with potential content problems, assuming that "rising content" wouldn't have any problem. 

I'll change this up.

codingisforlosers commented on models/admin/domains_proc.sql:
```diff
@@ -11,7 +11,7 @@ FROM (
 	google_analytics_account,
 	time_of_entry,
-	first_value(time_of_entry) OVER (PARTITION BY site_name ORDER BY time_of_entry DESC) lv
+	first_value(time_of_entry) OVER (PARTITION BY site_name,google_analytics_account,google_search_console_account ORDER BY time_of_entry DESC) lv
```
Not sure we want to partition at this granularity - we want to pull the latest value for the site as a whole (with this GA account / GSC account partition, if we update the account name it'd create two rows for the site).

codingisforlosers commented on models/visualization/actions_diffs_count.sql:
```diff
@@ -140,7 +140,7 @@ internal_links_out_count_diff,
 top_10_keywords_diff
 FROM {{ ref('actions_diffs') }}
-WHERE content_action not like 'Rising%' and content_action not like 'Below%'
+WHERE content_trajectory not like 'Rising%' and content_action not like 'Below%'
```
Let's revisit this filter - don't think we need this `WHERE` parameter (probably just the `AND`s).

codingisforlosers commented on models/base/ga/ga_stats.sql:
```diff
@@ -64,10 +64,6 @@ sum(goal_completions_all_goals_ttm) as goal_completions_all_goals_ttm,
 max(goal_conversion_rate_all_goals_ttm) as goal_conversion_rate_all_goals_ttm,
 max(bounce_rate_ttm) bounce_rate_ttm,
 max(avg_seconds_on_site_ttm) as avg_seconds_on_site_ttm,
-CASE WHEN sum(transactions_30d) > 0 THEN sum(transactions_30d)
```
Didn't we decide on leaving blended conversions in the models (for use in action recommendations on a line-level), but also recalculating them as calculated fields in DS for aggregated reporting purposes?

codingisforlosers commented:
@Mranfevb92 left a handful of comments here into the weeds of the models - once those are resolved we're good to merge

codingisforlosers commented:
It’s highly unlikely / impossible that they would (since content actions usually imply 0 traffic), so we don’t need to filter for it explicitly.

—
David Krevitt
Commissioner, CodingIsForLosers

> On Nov 23, 2020, at 10:57 AM, Andres Viana <notifications@github.com> wrote:
> 
> ﻿
> @Mranfevb92 commented on this pull request.
> 
> In models/visualization/actions_count.sql:
> 
> > @@ -53,7 +53,7 @@ FROM (
>  	pct_of_organic_sessions_30d
>  	FROM {{ ref('actions_hierarchy') }}
>  	WHERE content_action != '' AND content_action is not null	
> -	AND content_action not like 'rising%'
> +	AND content_trajectory not like 'rising%'
> Oh ok, though that the purpose was to keep only the pages with potential content problems, assuming that "rising content" wouldn't have any problem.
> 
> I'll change this up.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.

Commit:
Latest Changes 23nov

Commit:
Update agg_traffic.sql

*removing a duplicate of med_goal_conversion_rate_30d
